artiq/language/environment.py

```diff
@@ -242,6 +250,8 @@ def __init__(self, managers_or_parent, *args, **kwargs):
         self.__in_build = True
         self.build(*args, **kwargs)
         self.__in_build = False
+        if isinstance(self.__argument_mgr, ProcessArgumentManager):
```

Comment:
Why the isinstance check here?
Wouldn't an empty method in the other object types be better?

Reply:
Oh right, I didn't think of this solution.